src/aiodynamo/client.py

```diff
@@ -1,70 +1,76 @@
-from enum import Enum, unique
-from functools import partial
-from itertools import chain
-from typing import Any, AsyncIterator, Dict, List, TypeVar, Union
-
-import attr
-from boto3.dynamodb.conditions import ConditionBase, ConditionExpressionBuilder
-from boto3.dynamodb.types import DYNAMODB_CONTEXT
-from botocore.exceptions import ClientError
-
-from .errors import EmptyItem, ItemNotFound, TableNotFound
-from .models import (
+from __future__ import annotations
+
+import asyncio
+import datetime
+from dataclasses import dataclass
+from typing import *
+
+from aiodynamo.errors import *
+from aiodynamo.models import (
     GlobalSecondaryIndex,
     KeySchema,
     KeySpec,
     KeyType,
     LocalSecondaryIndex,
-    ProjectionExpr,
     ReturnValues,
     Select,
     StreamSpecification,
     TableDescription,
     TableStatus,
+    ThrottleConfig,
     Throughput,
+    TimeToLiveDescription,
+    TimeToLiveStatus,
+    WaitConfig,
+)
+from yarl import URL
+
+from .credentials import Credentials
+from .errors import (
+    TableDidNotBecomeActive,
+    TableDidNotBecomeDisabled,
+    Throttled,
+    TimeToLiveStatusNotChanged,
+)
+from .expressions import (
+    Condition,
+    KeyCondition,
+    Parameters,
+    ProjectionExpression,
     UpdateExpression,
-    get_projection,
 )
+from .http.base import HTTP
+from .sign import signed_dynamo_request
```

Comment:
Hmmm... the AWS signing page extolls the virtues of signing the requests when sent over HTTP...
I wonder if signing is required over HTTPS though...
There's https://forums.aws.amazon.com/message.jspa?messageID=705800 which mentions how to send unsigned payloads correctly, but I've no clue if dynamo would accept such payloads... 🤔

Reply:
that's for S3 which uses very different authentication strategies. DynamoDB requires all requests to be signed.